driver/sql/sql.go

```diff
@@ -16,3 +18,45 @@ type (
 		QueryContext(ctx context.Context, query string, args ...interface{}) (*sql.Rows, error)
 	}
 )
+
+// AcquireConn will return a new connection
+func AcquireConn(ctx context.Context, db *sql.DB) (*sql.Conn, error) {
+	var (
+		timeoutCtx    context.Context
+		timeoutCancel context.CancelFunc
+	)
+
+	for i := 0; i < 3; i++ {
```

Comment:
Why do we have 3 here and not let's say 5 or something else?

Reply:
This is similar to https://github.com/golang/go/blob/master/src/database/sql/sql.go#L1390